### PR TITLE
Handle webhooks properly

### LINKED ISSUE #360: Cannot read property 'user' of null
```
Dec-28 05:54:01.449 [DiscordBot] verbose: Failed to send message into room. TypeError: Cannot read property 'user' of null
    at UserSyncroniser.<anonymous> (/build/src/usersyncroniser.js:203:32)
    at Generator.next (<anonymous>)
    at /build/src/usersyncroniser.js:7:71
    at new Promise (<anonymous>)
    at __awaiter (/build/src/usersyncroniser.js:3:12)
    at UserSyncroniser.GetUserStateForGuildMember (/build/src/usersyncroniser.js:201:16)
    at UserSyncroniser.<anonymous> (/build/src/usersyncroniser.js:104:38)
    at Generator.next (<anonymous>)
    at /build/src/usersyncroniser.js:7:71
    at new Promise (<anonymous>)
    at __awaiter (/build/src/usersyncroniser.js:3:12)
    at UserSyncroniser.JoinRoom (/build/src/usersyncroniser.js:103:16)
    at DiscordBot.<anonymous> (/build/src/bot.js:699:45)
    at Generator.throw (<anonymous>)
    at rejected (/build/src/bot.js:5:65)
```

### PR DESCRIPTION
**Pull request by Sorunome**
This PR handles webhooks properly.

It fixes #360, #359 and #256 

#287 is unneeded, as discord.js creates a `User` for the webhooks with the `id` being the webhook ID, which appears to be unique among user IDs (the beauty of discord snowflakes)

### CODE REVIEW

File: src/bot.ts [hunk 1]
```diff
@@ -99,7 +99,13 @@ export class DiscordBot {
         return this.channelSync;
     }
 
-    public GetIntentFromDiscordMember(member: Discord.GuildMember | Discord.User): Intent {
+    public GetIntentFromDiscordMember(member: Discord.GuildMember | Discord.User, webhookID?: string): Intent {
+        if (webhookID) {
+            // webhookID and user IDs are the same, they are unique, so no need to prefix _webhook_
+            const name = member instanceof Discord.User ? member.username : member.user.username;
+            // no need to escape the mxid as getIntentFromLocalpart does that alreaddy down the road
+            return this.bridge.getIntentFromLocalpart(`_discord_${webhookID}_${name}`);
```
Comment (Half-Shot) [inline src/bot.ts]: Oh eer, you need to do ```new Matrix(`_discord_${webhookID}_${name}`).userId```

Comment (Half-Shot) [reply src/bot.ts]: *MatrixUser even

Comment (Sorunome) [reply src/bot.ts]: It does that under the hood, soru checked the library extra

Comment (Half-Shot) [reply src/bot.ts]: Okay, if you are sure :)

Comment (Sorunome) [reply src/bot.ts]: actually wait

**Pull request merged**